packages/fxa-settings/src/models/integrations/base-integration.ts

```diff
@@ -96,7 +98,27 @@ export abstract class Integration<
     this.features = { ...this.features, ...features } as T;
   }
 
-  isSync(): boolean {
+  /**
+   * This type guard determines if the integration is for an OAuth flow, meaning
+   * OAuthIntegration _or_ an extension of OAuthIntegration (e.g. OAuthBrowserIntegration).
+   */
+  isOAuth(): this is OAuthIntegration {
+    return this instanceof OAuthIntegration;
```

Comment:
This creates a ciculr dependency, so I think it should be avoided. Integration > OauthIntegration > BaseIntegration > Integration. To avoid it, just pull the type guard into its own function as it was before.

A flag that indicates if the integration supports oauth makes sense, but I think that can just follow the pattern set by other flags like isSync or isDesktopSync where the child class is ultimately responsible for returning the correct vaule.

Reply:
Will do. I think in this case then, it makes sense to go with Ben's renaming suggestion. I left this as a function at first, but `isOAuthIntegration` accounting for two kinds of integrations, one being named "oauth" and the other an extension of "oauth" felt a little confusing.